lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_workspace_runtime.rb

```diff
@@ -52,13 +52,43 @@ def self.instantiate_with_user(uri, user, attrs)
       workspace = MiqAeWorkspaceRuntime.new(attrs)
       self.current = workspace
       workspace.instantiate(uri, user, nil)
+
+      if !uri.nil?
+        _scheme, _userinfo, _host, _port, _registry, _path, _opaque, query, _fragment = MiqAeUri.split(uri, "miqaedb")
+        args = MiqAeUri.query2hash(query)
+        miq_request_id = find_miq_request_id(args)
```

Comment:
At this point, you should have a workspace - can we pull the request id out of the workspace similar to how we do it everywhere else?

Reply:
I originally had it that way but some spec tests failed because the workspace wasn't defined